### ISSUE TITLE
consider-using-generator error message ignores default arguments

### ISSUE THREAD

**Issue by ioah86** [pylint-dev/pylint#8563]
### Bug description

The built-in `min` function of python has multiple ways of calling it, as described in the [documentation](https://docs.python.org/3/library/functions.html#min).

While this

```python
min([i for i in range(len(line)) if line[i] not in whitespaces])
```
does correctly trigger the pylint error, I am also getting the error for this
```python
min([i for i in range(len(line)) if line[i] not in whitespaces], default=0)
```

There is no other way for no.2, and I think that the linter just does not take this way of calling min into account.

### Command used

```shell
pylint src/hidden_type/util.py
```

### Pylint output

```shell
************* Module src.hidden_type.util
src/hidden_type/util.py:523:14: R1728: Consider using a generator instead 'min(i for i in range(len(line)) if line[i] not in whitespaces)' (consider-using-generator)

------------------------------------------------------------------
Your code has been rated at 9.96/10 (previous run: 9.92/10, +0.04)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.2
Python 3.10.10 (main, Mar  5 2023, 22:26:53) [GCC 12.2.1 20230201]
```

### OS / Environment

Linux my-x1-yoga 6.1.23-1-lts #1 SMP PREEMPT_DYNAMIC Thu, 06 Apr 2023 10:56:12 +0000 x86_64 GNU/Linux
(Arch Linux)

**Comment by jacobtylerwalls** [comment]
Thanks for opening the issue. The error should still be raised, as your list comprehension could be a generator instead:

```python
min((i for i in range(len(line)) if line[i] not in whitespaces), default=0)
```

But the error message isn't showing you that. It's:
 - missing the keyword arg default=0
 - and thus neglects to parenthesize the generator.

Leaving this open to improve the message.
